File: src/main/java/com/apmosys/employeeportal/mongodb/dto/KnowledgeHubSearchResultProject.java
package com.apmosys.employeeportal.mongodb.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class KnowledgeHubSearchResultProject {

    private String projectId;
    private String projectName;
    private List<KnowledgeHubSearchResultObject> knowledgeHubSearchResultObjectList;
    private List<KnowledgeHubSearchResultObjectField> knowledgeHubSearchResultObjectFields;
    private Integer limit;
    private Integer skip;
    private Long totalGroupCount;
    private Long totalProjectOccurenceCount;
}

File: src/main/frontend/src/app/models/KnowledgeHubSearchResultProject.ts
import { KnowledgeHubSearchResultObject } from "./KnowledgeHubSearchResultObject";
import { KnowledgeHubSearchResultObjectField } from "./KnowledgeHubSearchResultObjectField";
import { ProjectInsightFacetCategoryDTO } from "./projectInsightFacetCategoryDTO";

export class KnowledgeHubSearchResultProject {
    projectId: any;
    projectName: any;
    knowledgeHubSearchResultObjectList: KnowledgeHubSearchResultObject[];
    knowledgeHubSearchResultObjectFields: KnowledgeHubSearchResultObjectField[];
    skip: any;
    limit: any;
    currentObjectPage: number;
    objectPageSize: number;
    totalObjectPageSize: number;
    totalObjectOccurenceCount: number;
    totalProjectOccurenceCount: number;
    totalGroupCount: number;
}

export class KnowledgeHubSearchResultWrapper {
    knowledgeHubSearchResultProjectList: KnowledgeHubSearchResultProject[];
    skip: any;
    limit: any;
    totalCount: any;
    facetCategories: ProjectInsightFacetCategoryDTO[];
}